src/Compilers/VisualBasic/Portable/Symbols/ReferenceManager.vb

```diff
@@ -255,172 +256,156 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
+                    Dim explicitAssemblyData = referencedAssemblies.Insert(0, assemblyBeingBuiltData)
+
+                    ' Let's bind all the references and resolve missing one (if resolver is available)
+                    Dim hasCircularReference As Boolean
+                    Dim corLibraryIndex As Integer
+                    Dim implicitlyResolvedReferences As ImmutableArray(Of MetadataReference) = Nothing
+                    Dim implicitlyResolvedReferenceMap As ImmutableArray(Of ResolvedReference) = Nothing
+                    Dim allAssemblyData As ImmutableArray(Of AssemblyData) = Nothing
+
+                    Dim bindingResult() As BoundInputAssembly = Bind(explicitAssemblyData,
+                                                                     compilation.Options.MetadataReferenceResolver,
+                                                                     compilation.Options.MetadataImportOptions,
+                                                                     allAssemblyData,
+                                                                     implicitlyResolvedReferences,
+                                                                     implicitlyResolvedReferenceMap,
+                                                                     resolutionDiagnostics,
+                                                                     hasCircularReference,
+                                                                     corLibraryIndex)
+
+                    Debug.Assert(bindingResult.Length = allAssemblyData.Length)
+
+                    references = references.AddRange(implicitlyResolvedReferences)
+                    referenceMap = referenceMap.AddRange(implicitlyResolvedReferenceMap)
+
+                    Dim referencedAssembliesMap As Dictionary(Of MetadataReference, Integer) = Nothing
+                    Dim referencedModulesMap As Dictionary(Of MetadataReference, Integer) = Nothing
+                    Dim aliasesOfReferencedAssemblies As ImmutableArray(Of ImmutableArray(Of String)) = Nothing
+
+                    BuildReferencedAssembliesAndModulesMaps(
+                        references,
+                        referenceMap,
+                        modules.Length,
+                        referencedAssembliesMap,
+                        referencedModulesMap,
+                        aliasesOfReferencedAssemblies)
+
+                    ' Create AssemblySymbols for assemblies that can't use any existing symbols.
+                    Dim newSymbols As New List(Of Integer)
+
+                    For i As Integer = 1 To bindingResult.Length - 1 Step 1
+                        If bindingResult(i).AssemblySymbol Is Nothing Then
+                            ' symbol hasn't been found in the cache, create a new one
+                            Dim compilationData = DirectCast(allAssemblyData(i), AssemblyDataForMetadataOrCompilation)
```

Comment:
The name of the local is confusing. Consider getting rid of it, it is used only once.